src/Routing/Router.php

```diff
@@ -902,17 +902,15 @@ public static function parseNamedParams(ServerRequest $request, array $options =
             '2.x backwards compatible named parameter support will be removed in 4.0'
         );
         $options += ['separator' => ':'];
-        if (empty($request->params['pass'])) {
-            $request->params['named'] = [];
-
-            return $request;
+        if (!$request->getParam('pass')) {
+            return $request->withParam('named', []);
```

Comment:
```‍‍‍php
$request->params['named'] = [];
return $request;
```
is not equal to 
```php
return $request->withParam('named', []);
```
i mean 
```php
return $request=$request->withParam('named', []);
```

Reply:
How are they not equivalent? In both cases the mutated request is returned.

Reply:
The difference is the request is not mutated in place. I've updated the tests to use the return value and will update the migration guide as well.